nabs/callbacks.py

```diff
@@ -165,3 +165,68 @@ def stop(self, doc):
                 tags=['run_table', 'RE'],
                 title='run_table'
             )
+
+
+class BECOptionFactoryFaker():
+    """
+    Takes an instance of the BestEffortCallback, and watches for metadata
+    keys to configure the BEC for a single run
+
+    Fakes a factory function to pass into RunRouter, for the purpose
+    of generating callbacks for a RunEngine.
```

Comment:
What exactly is being faked here? This looks very real to me

Reply:
Then normal pattern is to pass a factory function (`Callable[[str, dict], List[Callbacks], List[Subfactories]]`) to the RunRouter, rather than a class.  I guess it's enough like a duck to call a duck though

Reply:
Gotcha, I see the rationale. I think it would have been ok to just call it a `BECOptionFactory` but it's fine either way.